SU-HSDO/suhumsci
# Stn 660  caption credit refinements

# READY FOR REVIEW

## Summary
Further refine the Caption/Credit Styles considering H&S feedback to help us better utilize the view mode for additional elements and accommodate client content.

1. Address the bug where hb-raised-cards pushes all the captions to the top of the image
3. Update text to use Smart Trimmed
5. Update the text to span a wider area of the image to better accommodate  additional image use cases (postcards, etc.) Currently ~1/3, increase to ~1/2 (w/Heather’s discretion) 

[Jira Card: # 660](https://sparkbox.atlassian.net/browse/STN-660)

##Approach
Using the padding for the width of the caption allows us to keep the caption container full width for the gradient while controlling the length of the caption text.

Adding height: auto addresses hb-raised-cards bug with captions and gives it possible height decleration with any possible future utility classes that alter the cards image height.

## Need Review By (Date)
asap

## Urgency
'medium'

## Steps to Test
1. run `npm run test`
2. To test this card you'll want to use the test page with postcards and caption/credits created for testing STN-628.
3. You'll want to review the use of padding on the caption container to control the width of the caption.
4. Make sure your caption text limit is using smart trim and is set to 120 characters.
- You should be able to import this setting from my branch using ` lando drush cim`
5. Last, you'll want to add .hb-raised-cards to the component area with layout builder to make sure the captions are staying at the bottom and laying out correctly as they should.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
